### PR TITLE
Fail hard if the API returns an error

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: plugins/modules/session_request.py [hunk 1]
```diff
@@ -64,21 +64,18 @@ def _json_dumpable_dict(a_dict):
...
             if isinstance(result, DuffyAPIErrorModel):
-                session = None
-            else:
-                session = result.session.id
-                nodes = [_json_dumpable_dict(dict(node)) for node in result.session.nodes]
-                changed = True
+                self.fail_json(error=str(result))
+
+            session = result.session.id
+            nodes = [_json_dumpable_dict(dict(node)) for node in result.session.nodes]
```
Comment (ekohl) [inline plugins/modules/session_request.py:73]: I considered checking `if quantity and len(nodes) != quantity`. Not sure how much we should (dis)trust the Duffy API.

File: plugins/modules/session_request.py [hunk 2]
```diff
@@ -64,21 +64,18 @@ def _json_dumpable_dict(a_dict):
...
-        nodes = []
         if not self.check_mode:
             result = self.client.request_session([{'pool': self.params.get('pool'), 'quantity': self.params.get('quantity')}])
             if isinstance(result, DuffyAPIErrorModel):
-                session = None
-            else:
-                session = result.session.id
-                nodes = [_json_dumpable_dict(dict(node)) for node in result.session.nodes]
-                changed = True
+                self.fail_json(error=str(result))
```
Comment (evgeni) [inline plugins/modules/session_request.py]:
```suggestion
                self.fail_json(msg=str(result))
```

```
fail_json() missing 1 required positional argument: 'msg'
```

:kissing_cat:

File: tests/integration/targets/test_session_api_fail/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  session_request:
+    pool: does-not-exist
+  register: session
+  ignore_errors: True
+
+- name: assert failure
+  assert:
+    that:
+      - session is failed
+      - session.error is defined
```
Comment (evgeni) [inline tests/integration/targets/test_session_api_fail/tasks/main.yml]: this won't be defined anymore, but `.msg` will, I guess

**Comment by ekohl** [comment]
In the settings you can change the approval required to only recent GH accounts rather than people who have already contributed.

**Review by evgeni — APPROVED** [review]

**Pull request merged**